packages/react/src/lib/factories.ts

```diff
@@ -207,22 +215,25 @@ function createShorthandFromValue({
   // ----------------------------------------
   // Build up props
   // ----------------------------------------
-  const { defaultProps = {} } = options
+  const defaultProps = options ? options.defaultProps : undefined
 
   // User's props
   const usersProps = (valIsReactElement && {}) || (valIsPropsObject && (value as Props)) || {}
 
   // Override props
-  let { overrideProps } = options
+  let overrideProps = options ? options.overrideProps : undefined
   overrideProps =
     typeof overrideProps === 'function'
-      ? (overrideProps as Function)({ ...defaultProps, ...usersProps })
+      ? ((overrideProps as unknown) as Function)({ ...defaultProps, ...usersProps })
       : overrideProps || {}
 
   // Merge props
   const props = { ...defaultProps, ...usersProps, ...overrideProps }
```

Comment:
```suggestion
  const props: Props<P> = { ...defaultProps, ...usersProps, ...overrideProps }
```

Reply:
P doesn't have `className`, `style`, `styles`, `key` though

Reply:
It's `Props<P>`, `Props<P> = P & any` 🤔

Reply:
Sorry I meant `Props<P>`. I cannot do `props.className` if `const props: Props<P>`